cypress/e2e/supervision/deputies/add-deputy.cy.js

```diff
@@ -0,0 +1,93 @@
+        cy.get('[label="First name"]').type("Patrick");
+        cy.get('[label="Last name"]').type("Star");
+        cy.contains("Save & continue").click();
+        cy.waitForStableDOM();
+        cy.get('.footer > .dotted-link').click();
+        cy.get('.TABS_DEPUTIES').click();
+        cy.get('tr.summary-row > :nth-child(1) > .dotted-link').click();
+        cy.get('.person-name').should('contain.text', "Patrick Star");
+        cy.get('.summary-row.open > :nth-child(1)').should('contain.text', "Patrick Star");
+        cy.get('.deputy-details-type').should('contain.text', "Lay");
+        cy.get('.deputy-relation-type').should('contain.text', "Lay");
+        cy.get('.fee-payer').should('be.visible');
+        cy.get('.main-contact').should('be.visible');
+        cy.get('.order-details-main-correspondent').should('contain.text', 'Yes');
+        cy.get('.order-details-fee-payer').should('contain.text', 'Yes');
+      });
+    });
+
+    it("Greys out save and continue button when mandatory form fields not filled", () => {
+      cy.get("@clientId").then((clientId) => {
+        cy.searchForADeputyToReachAddADeputyPage();
+        cy.contains("Lay").click();
+        cy.get(':nth-child(1) > .radio-button').click();
+        cy.get('[label="First name"]').type("Squidward");
+        cy.get('.footer > :nth-child(1) > .button').should('be.disabled');
+      });
+    });
+
+    it("Does not allow over 200 characters in occupation field and shows validation summary", () => {
+      cy.get("@clientId").then((clientId) => {
+        cy.searchForADeputyToReachAddADeputyPage();
+        cy.contains("Lay").click();
+        cy.get('[label="First name"]').type("Sandy");
+        cy.get('[label="Last name"]').type("Squirrel");
+        cy.contains("Save & continue").click({force: true});
+        cy.get('[label="Occupation"]').type('abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrs');
+        cy.contains("Save & continue").click({force: true});
+        cy.get('.validation-summary').should('be.visible');
+        cy.get('.validation-summary').should('contain.text', "Occupation - The input is more than 200 characters long")
+      });
+    });
+
+    it("Allows a new fee payer to be set for an order", () => {
+      cy.get("@clientId").then((clientId) => {
+        cy.get('.TABS_DEPUTIES').click();
+        cy.get("@orderId").then((orderId) => {
+          cy.createADeputyAndAssignToExistingOrder(orderId);
+        });
+
+        // I can create a second deputy to set them as feepayer
+        cy.searchForADeputyToReachAddADeputyPage();
+
+        //check Lay type deputy
+        cy.get(':nth-child(1) > .radio-button').click();
+        cy.get('[label="First name"]').type("Kermit");
+        cy.get('[label="Last name"]').type("Frog");
+        cy.contains("Save & continue").click();
```

Comment:
Why the double click of save an continue?

Reply:
It's 3 pages, we fill in info on the first one, click save, click save again to finish the second form (where we amend nothing), set them as feepayer (hence the click) then save to finish the whole form.

Reply:
Ah yes it's this set, I remember them from looking at the old tests. Worth adding an assert or two to make sure the right intermediate page is loading then as otherwise it looks odd to anybody coming in in 6 months and they might be tempted to delete the duplicates.